lib/src/signed_video_openssl.c

```diff
@@ -62,6 +62,7 @@ typedef struct {
  */
 typedef struct {
   EVP_MD_CTX *ctx;  // Hashing context
+  EVP_MD_CTX *f_ctx;  // Hashing context for fallback to gop level hashing.
```

Comment:
I find the naming very misleading. "fallback to gop level" is a feature or an operation that may occur and has nothing to do with OpenSSL. I think you should focus on, throughout this PR, the difference between a "hash" and a "gop_hash".
So I suggest you name these context `hash_ctx` and `gop_hash_ctx`, or more generic `primary_ctx` and `secondary_ctx`, which is even better since OpenSSL is not aware of how they are used in the rest of the code.